File: ts-sdk/src/sdk/models/operations/listdeployrequests.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class ListDeployRequestsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class ListDeployRequestsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}

export class ListDeployRequestsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ListDeployRequestsPathParams;

  @SpeakeasyMetadata()
  queryParams: ListDeployRequestsQueryParams;
}

export class ListDeployRequestsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  listDeployRequests200ApplicationJSONObject?: Record<string, Record<string, any>>;
}

File: go-sdk/pkg/models/operations/listdeployrequests.go
package operations

type ListDeployRequestsPathParams struct {
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type ListDeployRequestsQueryParams struct {
	Page    *float64 `queryParam:"style=form,explode=true,name=page"`
	PerPage *float64 `queryParam:"style=form,explode=true,name=per_page"`
}

type ListDeployRequestsRequest struct {
	PathParams  ListDeployRequestsPathParams
	QueryParams ListDeployRequestsQueryParams
}

type ListDeployRequestsResponse struct {
	ContentType                                string
	StatusCode                                 int
	ListDeployRequests200ApplicationJSONObject map[string]map[string]interface{}
}
